src/app/contact/contact.component.html

```diff
@@ -0,0 +1,174 @@
+<div class="container">
+  <div style="margin-top: 50px; margin-left: 20%; margin-right: 20%;">
+    <div class="panel panel-default">
+      <div class="panel-heading">
+        <h3 style="text-align: center;">Contact Form</h3>
+      </div>
+      <div class="panel-body">
+        <div class="row">
+          <div class="col-md-8 offset-md-2">
+            <form novalidate (ngSubmit)="Save()" [formGroup]="contactForm">
+              <div class="form-group">
+                <label for="firstNameId">Title</label>
+                <div>
+                  <select class="form-control" id="title" formControlName="title">
+                    <option *ngFor="let title of titles" [value]="title">{{
+                      title
+                    }}</option>
+                  </select>
+                </div>
+              </div>
+              <div class="form-group">
+                <label for="firstNameId">First Name</label>
+                <div>
+                  <input class="form-control" id="firstNameId" type="text" placeholder="First Name (required)"
+                    formControlName="firstName" [ngClass]="{
+                      'is-invalid':
+                        (contactForm.get('firstName').touched ||
```

Comment:
Consider not calling the `get` methods of the form here. You can define getter methods in your component which you can use to access the validity of the form control from here. i.e.

in ContactComponent

```typescript

get firstName() { return this.contactForm.get('firstName'); }

```

Then HTML

```html
<input [ngClass]="is-invalid: firstName.touched ...">
```

Reply:
@strizzwald fixed